File: multiple-languages/java/ros-cdk-core/src/main/java/com/aliyun/ros/cdk/core/FnGetAZs.java
package com.aliyun.ros.cdk.core;

/**
 * The intrinsic function <code>Fn::GetAZs</code> returns an array that lists Availability Zones for a specified region.
 * <p>
 * Because customers have access to different Availability Zones, the intrinsic
 * function <code>Fn::GetAZs</code> enables template authors to write templates that adapt to the calling
 * user's access. That way you don't have to hard-code a full list of Availability Zones for a
 * specified region.
 */
@javax.annotation.Generated(value = "jsii-pacmak/1.85.0 (build 08ee592)", date = "2024-06-21T05:59:07.049Z")
@software.amazon.jsii.Jsii(module = com.aliyun.ros.cdk.core.$Module.class, fqn = "@alicloud/ros-cdk-core.FnGetAZs")
public class FnGetAZs extends com.aliyun.ros.cdk.core.Intrinsic {

    protected FnGetAZs(final software.amazon.jsii.JsiiObjectRef objRef) {
        super(objRef);
    }

    protected FnGetAZs(final software.amazon.jsii.JsiiObject.InitializationMode initializationMode) {
        super(initializationMode);
    }

    /**
     * Creates an <code>Fn::GetAZs</code> function.
     * <p>
     * @param region The name of the region for which you want to get the Availability Zones. This parameter is required.
     */
    public FnGetAZs(final @org.jetbrains.annotations.NotNull java.lang.Object region) {
        super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
        software.amazon.jsii.JsiiEngine.getInstance().createNewObject(this, new Object[] { region });
    }
}

File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Core/AlibabaCloud/SDK/ROS/CDK/Core/FnGetAZs.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Core
{
    /// <summary>The intrinsic function ``Fn::GetAZs`` returns an array that lists Availability Zones for a specified region.</summary>
    /// <remarks>
    /// Because customers have access to different Availability Zones, the intrinsic
    /// function <c>Fn::GetAZs</c> enables template authors to write templates that adapt to the calling
    /// user's access. That way you don't have to hard-code a full list of Availability Zones for a
    /// specified region.
    /// </remarks>
    [JsiiClass(nativeType: typeof(AlibabaCloud.SDK.ROS.CDK.Core.FnGetAZs), fullyQualifiedName: "@alicloud/ros-cdk-core.FnGetAZs", parametersJson: "[{\"docs\":{\"summary\":\"The name of the region for which you want to get the Availability Zones.\"},\"name\":\"region\",\"type\":{\"primitive\":\"any\"}}]")]
    public class FnGetAZs : AlibabaCloud.SDK.ROS.CDK.Core.Intrinsic
    {
        /// <summary>Creates an ``Fn::GetAZs`` function.</summary>
        /// <param name="region">The name of the region for which you want to get the Availability Zones.</param>
        public FnGetAZs(object region): base(_MakeDeputyProps(region))
        {
        }

        [System.Runtime.CompilerServices.MethodImpl(System.Runtime.CompilerServices.MethodImplOptions.AggressiveInlining)]
        private static DeputyProps _MakeDeputyProps(object region)
        {
            return new DeputyProps(new object?[]{region});
        }

        /// <summary>Used by jsii to construct an instance of this class from a Javascript-owned object reference</summary>
        /// <param name="reference">The Javascript-owned object reference</param>
        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
        protected FnGetAZs(ByRefValue reference): base(reference)
        {
        }

        /// <summary>Used by jsii to construct an instance of this class from DeputyProps</summary>
        /// <param name="props">The deputy props</param>
        [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
        protected FnGetAZs(DeputyProps props): base(props)
        {
        }
    }
}
